model/dependencies.go

```diff
@@ -90,8 +99,24 @@ func (di *dependencyIncluder) handle(pair TVPair) (bool, error) {
 
 	// queue up all dependencies for recursive inclusion
 	deps := di.expandDependencies(pair, bvt.DependsOn)
+
+	// If this function is invoked from generate.tasks, calculate all variants
+	// that need to be included as dependencies, but are not in the generated project.
+	// If all the task / variant pairs that spawn these dependencies are inactive, we
+	// also mark this newly generated variant as inactive.
+	pairSpecifiesActivation := activationInfo != nil && activationInfo.taskOrVariantHasSpecificActivation(pair.Variant, pair.TaskName)
 	for _, dep := range deps {
-		ok, err := di.handle(dep)
+		if activationInfo != nil && !activationInfo.variantExistsInGeneratedProject(dep.Variant) {
+			// If the new variant has not yet been added to deactivateGeneratedVariantTasks, or if the
```

Comment:
This double if statement is pretty hard to read; i think it might be good to break to a helper function that has a comment per line instead. 
variantExistsInGenerated can check if activationInfo == nil also.

I'm confused about why at 109 we just look at variant but I'm slightly confused about what's going on here. If a task depends on a variant, do we add all of those variants tasks to the depends on? (In which case I would think we'd want to keep track of the tasks instead)

Reply:
Moved to helper function.

True, there could be a case where some generated dependencies in a newly generated variant need to be activated, while others remain deactivated (added this case to unit tests). I suppose I missed that since server's original issue was fixed by deactivating the entire new variant. I've updated the logic here to track individual task/variant pairs rather than entire variants for better granularity.